### PR TITLE
Add local-2fa extension

### PR DESCRIPTION
**Pull request by vayaSEO**
## Description

**Local 2FA Codes** — fully offline, local-first 2FA TOTP code generator for Raycast on macOS.

### What it does
- Generates RFC 6238 TOTP codes (SHA1 / SHA256 / SHA512, 6 or 8 digits, configurable period)
- Stores accounts locally, encrypted with **AES-256-GCM** + **PBKDF2-SHA256** (600,000 iterations, OWASP 2023)
- Master password kept in Raycast `password` preference (macOS Keychain)

### Commands
- **List Codes** — view, copy, paste, delete accounts
- **Add Account** — unified flow: paste `otpauth://` URL with autofill, or fill manually
- **Import Google Migration** — paste `otpauth-migration://...` URLs
- **Import from QR Image** — select local PNG QR images (decoded in pure JS via `qr` + `pngjs`)

### Why a new extension
No existing Raycast extension offers fully offline TOTP with local encrypted storage, otpauth URL + Google Authenticator migration + QR image import in one place. All competing extensions rely on cloud services or external password managers.

### Security
- AES-256-GCM authenticated encryption, random 16-byte salt + 12-byte IV per save
- PBKDF2-SHA256 600k iterations (transparent backward-compat with legacy 210k vaults)
- In-process write mutex prevents concurrent-write races
- Input caps on Base32 / otpauth / migration parsers to prevent DoS
- Generic crypto errors (no leaks)
- `Clipboard.copy` uses `concealed: true`
- **No network calls. No external binaries. No `child_process`.**

Full threat model in [SECURITY.md](https://github.com/vayaSEO/local-2fa/blob/main/SECURITY.md).

### Tests
19 unit tests covering TOTP RFC 6238 vectors, `parseOtpauthUrl` edge cases, `parseGoogleMigrationUrl` edge cases, and `decodeBase32` bounds. Run with `npm test`.

## Screencast

Screenshots are included in `extensions/local-2fa/metadata/` (6 images at 2000×1250).

## Checklist

- [x] I read the [extension guidelines](https://developers.raycast.com/basics/prepare-an-extension-for-store)
- [x] I read the [documentation about publishing](https://developers.raycast.com/basics/publish-an-extension)
- [x] I ran `npm run build` and tested this distribution build in Raycast
- [x] I checked that files in the `assets` folder are used by the extension itself
- [x] I checked that assets used by the `README` are placed outside of the `metadata` folder

### CODE REVIEW

**Comment by vayaSEO** [comment]
Addressed in 5bc575e: algorithm mapping fixed (1=SHA1, 2=SHA256, 4=SHA512, 3=SHA384 unsupported), default algorithm changed to 1, category set to Security, ESLint config wrapped with defineConfig, CHANGELOG header pdated to [Initial Release].

File: extensions/local-2fa/src/google-migration.ts [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+
+  return {
+    issuer: issuer.trim() || "Unknown",
+    account: account.trim(),
+    secretBase32: toBase32(secret),
+    digits: digitsFromEnum(digits),
+    period: 30,
+    algorithm: algorithmFromEnum(algorithm),
+  };
+}
```
[the remark this answers is not in the document]
Comment (vayaSEO) [reply extensions/local-2fa/src/google-migration.ts:142]: Fixed in 0fc69db — algorithmFromEnum exception is now caught in parseOtpParameters and returns null to skip the entry, matching the existing pattern for HOTP / missing-secret cases. Added test coverage for the mixed-payload scenario (valid + unsupported in same payload).

File: extensions/local-2fa/src/storage.ts [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+  const cipher = createCipheriv("aes-256-gcm", key, iv);
+  const plaintext = Buffer.from(JSON.stringify(accounts), "utf8");
+  const ciphertext = Buffer.concat([cipher.update(plaintext), cipher.final()]);
+  const tag = cipher.getAuthTag();
+
+  return {
+    version: 2,
+    salt: salt.toString("base64"),
+    iv: iv.toString("base64"),
+    ciphertext: ciphertext.toString("base64"),
```
[the remark this answers is not in the document]
Comment (vayaSEO) [reply extensions/local-2fa/src/storage.ts:74]: Fixed in 8c10c1f — switched deriveKey/encryptAccounts/decryptAccounts to async using promisify(pbkdf2). PBKDF2 now runs on the libuv thread pool, so the event loop is no longer blocked and the Raycast UI does not freeze during save/delete. Kept the fresh-salt-per-write behaviour for crypto hygiene; the cache benefits the read path. SECURITY.md and CHANGELOG.md updated to reflect this.

**Review by pernielsentikaer — COMMENTED** [review]
Hi 👋

There are already several extensions in the store with that purpose.

The closest to "local 2FA" is Two-Factor Authentication Code Generator by cjdenio, which generates TOTP codes locally with secrets in Raycast's encrypted database: https://www.raycast.com/cjdenio/two-factor-authentication-code-generator

Other local-2FA options in the store:
• OTP Auth: https://www.raycast.com/huzef44/otp-auth
• Ente Auth: https://www.raycast.com/chkpwd/ente-auth
• Proton Authenticator (reads Proton's local DB): https://www.raycast.com/Fared/proton-authenticator
• Aegis Authenticator: https://www.raycast.com/bfayers/aegis
• 2FA Code Finder (from iMessage): https://www.raycast.com/yuercl/imessage-2fa

Coould you check them out?
